src/utils/stark/starkCurve.ts

```diff
@@ -100,3 +109,58 @@ export function generateStarkPrivateKey(): string {
   const keyPair = starkEc.genKeyPair();
   return grindKey(keyPair.getPrivate(), starkEcOrder);
 }
+
+function getIntFromBits(
+  hex: string,
+  start: number,
+  end: number | undefined = undefined,
+): number {
+  const bin = encUtils.hexToBinary(hex);
+  const bits = bin.slice(start, end);
+  const int = encUtils.binaryToNumber(bits);
+  return int;
+}
+
+function getAccountPath(
+  layer: string,
+  application: string,
+  ethereumAddress: string,
+  index: string,
+): string {
+  const layerHash = hash.sha256().update(layer).digest('hex');
+  const applicationHash = hash.sha256().update(application).digest('hex');
+  const layerInt = getIntFromBits(layerHash, -31);
+  const applicationInt = getIntFromBits(applicationHash, -31);
+  const ethAddressInt1 = getIntFromBits(ethereumAddress, -31);
+  const ethAddressInt2 = getIntFromBits(ethereumAddress, -62, -31);
+  return `m/2645'/${layerInt}'/${applicationInt}'/${ethAddressInt1}'/${ethAddressInt2}'/${index}`;
+}
+
+function getKeyFromPath(seed: string, path: string): string {
+  const privateKey = hdkey
+    .fromMasterSeed(Buffer.from(seed.slice(2), 'hex')) // assuming seed is '0x...'
+    .derivePath(path)
+    .getWallet()
+    .getPrivateKey();
+  return grindKey(new BN(privateKey), starkEcOrder);
+}
+
+/**
+ * Generates a deterministic Stark private key from the provided signer.
+ * @returns the private key as a hex string
+ */
+export async function generateLegacyStarkPrivateKey(
```

Comment:
I think the RFC document would be a good place to suggest a rename but I would have to point out that the Typescript SDK exports everything from @imtbl/core-sdk path and doesn't have the benefit of namespaces so if we just name it LegacyKey it may not be descriptive enough.